Assets/SteamVR_Unity_Toolkit/Scripts/VRTK_PlayerClimb.cs

```diff
@@ -0,0 +1,188 @@
+            if (PlayerClimbStarted != null)
+                PlayerClimbStarted(this, e);
+        }
+
+        public virtual void OnPlayerClimbEnded(PlayerClimbEventArgs e)
+        {
+            if (PlayerClimbEnded != null)
+                PlayerClimbEnded(this, e);
+        }
+
+        protected PlayerClimbEventArgs SetPlayerClimbEvent(uint controllerIndex, GameObject target)
+        {
+            PlayerClimbEventArgs e;
+            e.controllerIndex = controllerIndex;
+            e.target = target;
+            return e;
+        }
+
+        private void Start()
+        {
+            var controllerManager = GameObject.FindObjectOfType<SteamVR_ControllerManager>();
+            InitControllerListeners(controllerManager.left);
+            InitControllerListeners(controllerManager.right);
+
+            // Listen for teleport events 
+            VRTK_BasicTeleport teleportComponent = GetComponent<VRTK_BasicTeleport>();
+            if (teleportComponent)
+            {
+                teleportComponent.Teleporting += new TeleportEventHandler(OnTeleport);
+            } 
+
+            // Required Components
+
+            // VRTK_PlayerPhysics
+            playerPhysics = GetComponent<VRTK_PlayerPhysics>();
+            if (!playerPhysics && useGravity)
+            {
+                playerPhysics = gameObject.AddComponent<VRTK_PlayerPhysics>();
+            }
+
+            // VRTK_HeadsetCollisionFade
+            Transform headCamera = GameObject.FindObjectOfType<SteamVR_GameView>().GetComponent<Transform>();
+            VRTK_HeadsetCollisionFade collisionFade = headCamera.GetComponent<VRTK_HeadsetCollisionFade>();
+            if (collisionFade == null)
+            {
+                collisionFade = headCamera.gameObject.AddComponent<VRTK_HeadsetCollisionFade>();
+            }
+            collisionFade.HeadsetCollisionDetect += new HeadsetCollisionEventHandler(OnHeadsetCollisionDetected);
+            collisionFade.HeadsetCollisionEnded += new HeadsetCollisionEventHandler(OnHeadsetCollisionEnded);
+        }
+
+        private Vector3 GetPosition(Transform objTransform)
+        {
+            if (usePlayerScale)
+            {
+                return Vector3.Scale(objTransform.localPosition, gameObject.transform.localScale);
+            }
+
+            return objTransform.localPosition;
+        }
+
+    
+        private void OnGrabObject(object sender, ObjectInteractEventArgs e)
+        {
+            VRTK_InteractableObject interactObject = e.target.GetComponent<VRTK_InteractableObject>();
+            if( interactObject && interactObject.AttachIsClimbObject() )
+            {
+                if (useGravity)
+                {
+                    playerPhysics.StopPhysicsFall();
+                }
+
+                OnPlayerClimbStarted(SetPlayerClimbEvent(e.controllerIndex, e.target));
+                isClimbing = true;
+                controllerTransform = ((VRTK_InteractGrab)sender).gameObject.transform;
```

Comment:
You may also be able to do just `transform` here not sure because it's coming form the sender.